mnefun/_epoching.py

```diff
@@ -146,7 +150,8 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
             write_hdf5(hdf5_file, use_reject, overwrite=True)
         else:
             use_reject = _handle_dict(p.reject, subj)
-        # create epochs
+        # read in events and create epochs
+        events = _read_events(p, subj, run_indices[si], raw, picker='restrict')
```

Comment:
Looking at this again, I think this is a backward compatibility break @NeuroLaunch, `_read_events` used to give back all events, no?

Reply:
Maybe it's actually a bugfix to do this actually...

Reply:
It's a bug fix at least for the way I set up my event list.  But I think it should be compatible with previous uses, as the line subsequent to 154 that creates the (final saved) epochs is limited to those in `event_id=old_dict`, which are the same ids enforced by 'restrict'.

Reply:
Also, in reviewing Ashley's soma scripts, she also has unused event ids in the stored event list.  So, implementing a restricted epoching during the cov step may give a better tmin:tmax estimate of covariance since only the intended events will be used.  (If the tmin:tmax (or is it bmin:bmax for covariance?) is particularly long, excessive events may lead to considerable redundancy in the time samples used to estimate covariance, which would lead to unwanted bias.)